### PR TITLE
Allow editing of gasPrice & gas in Signer

### PR DESCRIPTION
**Pull request by jacogr**
- Closes https://github.com/ethcore/parity/issues/2605
- There are still rough edges, but good as a first iteration to start getting a feel of how and see how to improve further

https://youtu.be/nLfi7kUD3o4

### CODE REVIEW

File: js/src/ui/Editor/mode-solidity.js [hunk 1]
```diff
@@ -22,6 +22,12 @@
 /* eslint-disable */
 var ace = window.ace;
 
+if (process.env.NODE_ENV === 'test') {
```
Comment (jacogr) [inline js/src/ui/Editor/mode-solidity.js]: `npm run test` fix - really not sure why it only pops up now and why the rest of the `window.ace` object is defined but not `define(...)`

File: js/src/views/Signer/components/TransactionMainDetails/transactionMainDetails.js [hunk 2]
```diff
@@ -14,52 +14,43 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
+import MapsLocalGasStation from 'material-ui/svg-icons/maps/local-gas-station';
```
Comment (ngotchac) [inline js/src/views/Signer/components/TransactionMainDetails/transactionMainDetails.js]: Nothing to do with this PR, but I was thinking about this:
What about have an Icons util somewhere in the UI ? We don't use that many icons, but still whenever I want to find the right one I end up going to design.google.com. There might even be some inconsistencies in the app (using different icons for the same purpose).

File: js/src/views/Signer/components/TransactionPendingForm/transactionPendingForm.js [hunk 3]
```diff
@@ -72,22 +71,23 @@ export default class TransactionPendingForm extends Component {
     if (!isRejectOpen) {
       html = <span>reject transaction</span>;
     } else {
-      html = <span><BackIcon />I've changed my mind</span>;
+      html = <span><BackIcon />{ "I've changed my mind" }</span>;
```
Comment (ngotchac) [inline js/src/views/Signer/components/TransactionPendingForm/transactionPendingForm.js]: :+1:

**Comment by ngotchac** [comment]
Looks good, finally got that in there ! :)

A few things though:

- `Failed prop type: The prop `gasLimit` is marked as required in `RequestPending`, but its value is `undefined`` on signer load
- `Error: gte() not a number: undefined` in `GasPriceEditor.__setGas` when modifying provided gas
- `Invalid prop `value` supplied to `Input`` in `GasPriceEditor` when modifying gas price
- The Gas Price chart looks a bit flat (whether too large or not tall enough)

**Comment by ngotchac** [comment]
Also, I got `Invalid Params` when modifying the gas prices / amount. Should I need a the rebuild Parity ?

File: js/src/views/Signer/components/TransactionMainDetails/transactionMainDetails.js (hunk 2 above)
Comment (jacogr) [inline js/src/views/Signer/components/TransactionMainDetails/transactionMainDetails.js]: Actually a good point, like it. Also conforms more to our approach of using our components and, as you mentioned, makes sure we have consistency.

**Comment by jacogr** [comment]
Yes. You need latest Parity - gas was not available via the RPC, it is since yesterday.

**Comment by jacogr** [comment]
The Props stuff got screwed somewhere in one of my merges. (I had some endless issues losing & conflicting changes each time I swapped branches to this one for some obscure reason.)

**Comment by jacogr** [comment]
Props re-re-added, fixed minor issue in transfer display

**Comment by ngotchac** [comment]
Everything is working fine.
Still got this one however:
```
Warning: Failed prop type: Invalid prop `value` supplied to `Input`.
    in Input (created by GasPriceEditor)
    in GasPriceEditor (created by TransactionPending)
```
when editing gas price from the slider/graphic selector

**Pull request merged**